### PR TITLE
upgrade to 0.8.27

### PR DESCRIPTION
**Pull request by livingrockrises**
https://soliditylang.org/blog/2024/09/04/solidity-0.8.27-release-announcement/

you can upgrade your solc version and test. https://github.com/ethereum/solidity/releases/tag/v0.8.27

### CODE REVIEW

**Comment by ankurdubey521** [comment]
Oh this is even better

File: foundry.toml
Comment (filmakarov) [inline foundry.toml:1]: I suggest `via-ir = false` :)

Compiling 205 files with Solc 0.8.27

via-ir = false: Solc 0.8.27 finished in 22.65s
via-ir = true: Solc 0.8.27 finished in 177.23s

**Review by filmakarov — CHANGES_REQUESTED** [review]

File: foundry.toml
Comment (livingrockrises) [reply foundry.toml:1]: where? I am setting env variable to false when I run

Comment (livingrockrises) [reply foundry.toml:1]: ah missed in foundry.toml

Comment (livingrockrises) [reply foundry.toml:1]: done

**Review by ankurdubey521 — APPROVED** [review]

**Review by filmakarov — APPROVED** [review]

File: foundry.toml [hunk 1]
```diff
@@ -6,13 +6,13 @@
   bytecode_hash = "none"
   evm_version = "paris"           # See https://www.evmdiff.com/features?name=PUSH0&kind=opcode
   fuzz = { runs = 1_000 }
-  via-ir = true
```
Comment (NoahMarconi) [inline foundry.toml:9]: might prefer profiles over globally turning off to allow for via-ir when deploying

Comment (NoahMarconi) [reply foundry.toml:9]: although I see you have it in the hardhat to toggle on/off

**Pull request merged**